packages/backend/migrations/20231121184537_add_resource_fkeys_webhook_event.js

```diff
@@ -32,6 +32,47 @@ exports.up = function (knex) {
         .index()
         .onDelete('CASCADE')
     })
+    .then(() => {
+      // Delete webhook events with data corresponding to a deleted entity (otherwise this migration will fail when populating the foreign keys in the next step)
+      return knex.raw(
+        `
+        DELETE FROM "webhookEvents" WHERE id IN (
+          SELECT "webhookEvents".id FROM "webhookEvents" 
+          LEFT JOIN "incomingPayments" ON ("webhookEvents".data->>'id')::uuid = "incomingPayments".id
+          WHERE "webhookEvents".type IN ('incoming_payment.created', 'incoming_payment.completed', 'incoming_payment.expired')
+          AND "incomingPayments".id IS NULL
+        );
+  
+        DELETE FROM "webhookEvents" WHERE id IN (
+          SELECT "webhookEvents".id FROM "webhookEvents"
+          LEFT JOIN "outgoingPayments" ON ("webhookEvents".data->>'id')::uuid = "outgoingPayments".id
+          WHERE "webhookEvents".type IN ('outgoing_payment.created', 'outgoing_payment.completed','outgoing_payment.failed')
+          AND "outgoingPayments".id IS NULL
+        );
+        
+        DELETE FROM "webhookEvents" WHERE id IN (
+          SELECT "webhookEvents".id FROM "webhookEvents"
+          LEFT JOIN "peers" ON ("webhookEvents".data->>'id')::uuid = "peers".id
+          WHERE "webhookEvents".type IN ('peer.liquidity_low')
+          AND "peers".id IS NULL
+        );
+        
+        DELETE FROM "webhookEvents" WHERE id IN (
+          SELECT "webhookEvents".id FROM "webhookEvents"
+          LEFT JOIN "assets" ON ("webhookEvents".data->>'id')::uuid = "assets".id
+          WHERE "webhookEvents".type IN ('asset.liquidity_low')
+          AND "assets".id IS NULL
+        );
+        
+        DELETE FROM "webhookEvents" WHERE id IN (
+          SELECT "webhookEvents".id FROM "webhookEvents"
+          LEFT JOIN "walletAddresses" ON ("webhookEvents".data->'walletAddress'->>'id')::uuid = "walletAddresses".id
+          WHERE "webhookEvents".type IN ('wallet_address.web_monetization')
+          AND "walletAddresses".id IS NULL
+        );
+        `
+      )
```

Comment:
The only realistic possibility is the incoming payments failing, since thats the only thing we hard delete regularly (we hard delete peers as well, but its unlikely a peer would be deleted after receiving `'peer.liquidity_low'` events). We don't hard delete outgoing payments, assets, or wallet addresses